pkg/doc/verifiable/embedded_proof_test.go

```diff
@@ -96,9 +112,26 @@ func Test_checkEmbeddedProof(t *testing.T) {
     "proofValue": "invalid value"
   }
 }`
-		docBytes, err := checkEmbeddedProof([]byte(docWithNotSupportedProof), defaultVCOpts)
+		docBytes, err := checkEmbeddedProof([]byte(docWithNotSupportedProof),
+			&credentialOpts{publicKeyFetcher: SingleKey([]byte("pub key bytes"), kms.ED25519)})
 		r.Error(err)
 		r.Contains(err.Error(), "check embedded proof")
 		r.Nil(docBytes)
 	})
+
+	t.Run("no public key fetcher defined", func(t *testing.T) {
+		docWithNotSupportedProof := `{
+  "@context": "https://www.w3.org/2018/credentials/v1",
+  "proof": {
+	"type": "Ed25519Signature2018",
+    "created": "2020-01-21T12:59:31+02:00",
+    "creator": "John",
+    "proofValue": "invalid value"
+  }
+}`
+		docBytes, err := checkEmbeddedProof([]byte(docWithNotSupportedProof), defaultVCOpts)
+		r.Error(err)
```

Comment:
out of curiosity, any reason of using a require instance `r` instead of `require` ?

why is this test function creating a new instance r?

Reply:
Well, I usually use in test functions with many asserts. It's more concise - `r` instead of `require`, and also we don't pass `t` as a first argument.